src/color.rs

```diff
@@ -32,6 +52,7 @@ impl Color {
 
     #[inline]
     pub fn as_argb8888(&self) -> u32 {
+        validate_color(self);
```

Comment:
Likewise, we can't exist and not be valid with the check in new.

Reply:
I've found that `new` isn't always called, particularly when loading from the config, hence why I sprinkled the validation check everywhere that would use the loaded `Color`.

(Note that this is basically my first time doing anything in rust, so I may be missing some things someone familiar with the language knows)